Sources/XcodeGenKit/PBXProjGenerator.swift

```diff
@@ -294,6 +335,67 @@ public class PBXProjGenerator {
         return targetDependency
     }
 
+    func generateExternalTargetDependency(from: String, to target: String, in project: String) -> (PBXTargetDependency, PBXTarget, PBXReferenceProxy) {
+        guard let projectReference = self.project.getProjectReference(project) else {
+            fatalError("project not found")
+        }
+
+        let pbxProj = try! getPBXProj(from: projectReference)
+
+        guard let targetObject = pbxProj.targets(named: target).first else {
+            fatalError("target not found")
+        }
+
+        let projectFileReferenceIndex = self.pbxProj.rootObject!
+            .projects
+            .map { $0["ProjectRef"] as? PBXFileReference }
+            .firstIndex { $0?.path == Path(projectReference.path).normalize().string }
+
+        guard let index = projectFileReferenceIndex,
+            let projectFileReference = self.pbxProj.rootObject?.projects[index]["ProjectRef"] as? PBXFileReference,
+            let productsGroup = self.pbxProj.rootObject?.projects[index]["ProductGroup"] as? PBXGroup else {
```

Comment:
This file is getting too large and complicated. Will have to look at splitting this up into things like `TargetGenerator` and the like, after this